zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js

```diff
@@ -865,6 +865,30 @@ angular.module('zeppelinWebApp').controller('ParagraphCtrl', function($scope, $r
     }
   };
 
+  $scope.parseTableCell = function(cell) {
+    //here we try to automatically parse the data type for each
+    //cell in the table, i.e. string/number/date
+    // 1. is this a date?
+    var supportedDateFormats = ['MM-DD-YYYY', 'DD-MM-YYYY', 'MM/DD/YYYY',
```

Comment:
Yeah my bad, and we are forced to provide a format since moment(String) is inconsistent depending on the web browser.

Since the Integer check is faster than the Date one, do you think we could make the Integer check before the Date?